ibis/expr/operations/window.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+
+from public import public
+
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.rules as rlz
+from ibis.expr.operations import Value
+
+
+@public
+class WindowBoundary(Value):
+    # TODO(kszucs): consider to prefer Concrete base class here
+    # pretty similar to SortKey and Alias operations which wrap a single value
+    value = rlz.one_of([rlz.numeric, rlz.interval])
+    preceding = rlz.bool_
+
+    output_shape = rlz.shape_like("value")
+    output_dtype = rlz.dtype_like("value")
+
+    @property
+    def name(self) -> str:
+        self.value.name
```

Comment:
```suggestion
        return self.value.name
```

and this probably needs a test to catch things like a missing `return` :)

Reply:
Nice catch! Actually we don't need to implement `name` here since `ops.Value` provides a default implementation.